### DISCUSSION TITLE
Extend AuditEventWebApi

### DISCUSSION THREAD

**Discussion by JBastiaan** [thepirat000/Audit.NET#503] [Q&A]
I've got a question of whats possible with the current library. We are planning on using the AuditApi attributes to audit incoming requests. We would like to be able to extend the AuditEventWebApi event to be able to easily add some additional fields to the event being saved. Currently we can also do it using 

`scope.SetCustomField(nameof(CreateSubscriptionAuditEvent.UserId), userId);`

It would be nice however to be able to do:
`scope.Event.UserId = userId;`

Is there a way to extend the AuditEventWebApi when using the AuditApi attribute?
I did a little digging and i don't see a way to override an ActionType so it will always add my custom eventtype, also the attribute news up an AuditEventWebApi by the looks of it. But maybe im missing something?

Thanks for the nice library!

**Comment by thepirat000** [comment]
Even if there was a way to extend the `AuditEventWebApi`, you will have to cast the `scope.Event` property, so it would be more like: `((YourCustomEventType)scope.Event).UserId = userId;` which probably isn't too nice.

You can also access the custom field dictionary directly:

```c#
scope.Event.CustomFields["UserId"] = userId;
```

Or even inside the WebApi Action:

```c#
(scope.Event as Audit.WebApi.AuditEventWebApi).Action.CustomFields["UserId"] = userId;
```

Anyway I'll think a little bit more about this

**Reply by thepirat000** [reply]
converting this to a discussion since it doesn't seem to be an issue per se
